### PR TITLE
Added Readme and env files

### PR DESCRIPTION
**Pull request by yash3886**
Completed both readmes and added the required env file in the source dir

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+```bash
+<path-to-opt> -custom_loop_distribution -cld-use-pipe-inf -cld-data-format=<data-format> -cld-pipe-name=<any-pipe-name> <path-to-the-ll-file> -ml-config-path=<path-to-the-config-file>
+```
+
+Note 1: In above the command supported data formats are "bytes" and "json"
+
+Note 2: The pipe name can be anything just make sure you use the same pipe name both the server and client side.
+
+
+
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Please add link to model's readme as in RL4ReAl and POSET-RL

File: model/LoopDistribution/Readme.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* ```--data_format json```:Uses Json as the dataformat
+* ```--data_format bytes```:Uses Bytes as the dataformat
+* ```--data_format protobuf```:Uses Protobuf as the dataformat
+
+## Training Logs
+Training logs are written in `~/ray_results` directory by default.
+- Customize the path using the following syntax in `experiment.py`
+    ```py
+    ray.init(_temp_dir="<path_to_raylog>")  
+    ```
```
Comment (svkeerthy) [inline model/LoopDistribution/Readme.md]: Please add link to inference's readme as in RL4ReAl and POSET-RL

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: model/LoopDistribution/Readme.md (hunk 2 above)
Comment (yash3886) [reply model/LoopDistribution/Readme.md]: Added in the new commit.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 1 above)
Comment (yash3886) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Added in the new commit.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+# Loop Distribution
+
+This directory contains the Python model related files for training and inferencing Loop Distribution.
+
+
+
+This repo contains the source code and relevant information described in the [paper](https://ieeexplore.ieee.org/abstract/document/10026979)
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl_loop_distribution/) for more details.
+
+> Reinforcement Learning assisted Loop Distribution for Locality and Vectorization, Shalini Jain, S. VenkataKeerthy, Rohit Aggarwal, Tharun Kumar Dangeti, Dibyendu Das, Ramakrishna Upadrasta
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Please add the venue and year of publication.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+## Environment Setup
+
+
+Setup the environment using the `ml-llvm-project/model/LoopDistribution/src/LOF_original_env.yml` file with the following command
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Clean this up. Directly say conda env create -f  file name.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+```bash
+conda env create -f <path-to-the-yml-file>
+```
+
+
+## Inference on Trained Models
+
+Currently three kinds of model runners are supported by us, via the `MLCompilerBridge`
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md:27]: Are all three working? Please talk only about Pipes and ONNX.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+## Inference on Trained Models
+
+Currently three kinds of model runners are supported by us, via the `MLCompilerBridge` 
+- [ONNX Model Runner](#inference-flow-using-onnx-model-runner) - In-Process Model Runner
+- [gRPC Model Runner](#inference-flow-using-grpc-model-runner) - gRPC Based Model Runner
+- [Pipe Model Runner](#inference-flow-using-pipe-model-runner) - Pipe based Model Runner
+
+`Note`: Run ONNX based Model Runner for inference flow to get light and fast compilation.
+
+## Inference flow using gRPC Model Runner
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md:34]: Is this flow working? Remove it if it is not.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+### Starting the gRPC Server:
+File `inference.py` under the path `ml-llvm-project\model\LoopDistribution\src\` contains script to start the servers.
```
Comment (svkeerthy) [inline llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Fix "\\" -> "/"

File: model/LoopDistribution/Readme.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+This repo contains the source code and relevant information described in the [paper](https://ieeexplore.ieee.org/abstract/document/10026979)
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl_loop_distribution/) for more details.
+
+> Reinforcement Learning assisted Loop Distribution for Locality and Vectorization, Shalini Jain, S. VenkataKeerthy, Rohit Aggarwal, Tharun Kumar Dangeti, Dibyendu Das, Ramakrishna Upadrasta
+
+
+## Environment Setup
+
+
+Setup the environment using the `ml-llvm-project/model/LoopDistribution/src/LOF_original_env.yml` using the following commands
```
Comment (svkeerthy) [inline model/LoopDistribution/Readme.md]: Clean this as well

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 6 above)
Comment (Yakshimist) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md:34]: Yes GRPC is working fine.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 5 above)
Comment (Yakshimist) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md:27]: All the three are working fine.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 7 above)
Comment (Yakshimist) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: okay

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 4 above)
Comment (Yakshimist) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Done.

File: llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md (hunk 3 above)
Comment (Yakshimist) [reply llvm/lib/Transforms/Scalar/IR2Vec-LOF/custom_loop_distribution/Readme.md]: Done

File: model/LoopDistribution/Readme.md (hunk 8 above)
Comment (yash3886) [reply model/LoopDistribution/Readme.md]: done

**Review by svkeerthy — APPROVED** [review]

**Pull request merged**